farmerbot/models/config.go

```diff
@@ -0,0 +1,226 @@
+package models
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/rs/zerolog/log"
+	substrate "github.com/threefoldtech/tfchain/clients/tfchain-client-go"
+	"github.com/threefoldtech/tfgrid-sdk-go/farmerbot/constants"
+	"github.com/threefoldtech/tfgrid-sdk-go/farmerbot/slice"
+)
+
+// InputConfig is the inputs for configuration for farmerbot
+type InputConfig struct {
+	FarmID        uint32   `json:"farm_id" yaml:"farm_id" toml:"farm_id"`
+	IncludedNodes []uint32 `json:"included_nodes" yaml:"included_nodes" toml:"included_nodes"`
+	ExcludedNodes []uint32 `json:"excluded_nodes" yaml:"excluded_nodes" toml:"excluded_nodes"`
+	Power         Power    `json:"power" yaml:"power" toml:"power"`
+}
+
+// Config is the configuration data for farmerbot
+type Config struct {
+	Farm  substrate.Farm `json:"farm" yaml:"farm" toml:"farm"`
+	Nodes []Node         `json:"nodes" yaml:"nodes" toml:"nodes"`
+	Power Power          `json:"power" yaml:"power" toml:"power"`
+	*sync.Mutex
```

Comment:
please make it `m sync.Mutex`, embedding mutex is dangerou as anyone with access to the struct can do mystruct.Lock, .. etc